### PR TITLE
Update kataconfig CR status on scaling of nodes

### PR DESCRIPTION
**Pull request by bpradipt**
**- Description of the problem which is fixed/What is the use case**
Output of kataconfig CR does not update when nodes are scaled up

**- What I did**
Watch for changes to MachineConfigPool and trigger reconcile 

**- How to verify it**
Add a new node to the existing cluster. Run `oc describe kataconfig example-kataconfig` to display the status of 
the CR

**- Description for the changelog**
Update kataconfig CR status on scaling of nodes

### CODE REVIEW

File: config/extension-crds/machineconfig.crd.yaml [hunk 1]
```diff
@@ -0,0 +1,341 @@
+apiVersion: apiextensions.k8s.io/v1
```
Comment (jensfr) [inline config/extension-crds/machineconfig.crd.yaml:1]: I'm surprised we have to keep a copy of the the CRD. Is this done automatically by operator-sdk? What happens if MCO changes something on one of the CRDs? Do we have to check the version of the CRD or something like that?

Comment (bpradipt) [reply config/extension-crds/machineconfig.crd.yaml:1]: Using the OpenShift CRDs with `envtest` looks a bit convoluted. 
It took me fair amount of time to figure out a working scenario. I'm sure there will be a better way.

Please have a look at the following issues - 
https://github.com/kubernetes-sigs/controller-runtime/issues/1191
https://github.com/operator-framework/operator-sdk/issues/4434
https://github.com/kubernetes-sigs/controller-runtime/pull/1393

We do run the risk of test cases breaking or giving false positives if MCO changes the CRD and we don't update the same for the operator tests.

Comment (jensfr) [reply config/extension-crds/machineconfig.crd.yaml:1]: I see. It's a known problem and this is the best we can do right now. Thanks!

**Comment by jensfr** [comment]
lgtm

**Review by jensfr — APPROVED** [review]

**Pull request merged**